lib/SILOptimizer/Mandatory/MandatoryInlining.cpp

```diff
@@ -404,17 +404,22 @@ static SILFunction *getCalleeFunction(
     return nullptr;
   }
 
-  // If CalleeFunction is a declaration, see if we can load it. If we fail to
-  // load it, bail.
-  if (CalleeFunction->empty()
-      && !AI.getModule().linkFunction(CalleeFunction, Mode))
-    return nullptr;
-
   // If the CalleeFunction is a not-transparent definition, we can not process
   // it.
   if (CalleeFunction->isTransparent() == IsNotTransparent)
     return nullptr;
 
+  // If CalleeFunction is a declaration, see if we can load it.
+  if (CalleeFunction->empty()) {
+    // FIXME: Remove 'Mode'
+    if (Mode != SILOptions::LinkingMode::LinkNone)
+      AI.getModule().loadFunction(CalleeFunction);
```

Comment:
I think this function has the wrong name. Imagine a naive reader... what is the difference in between loadFunction and linkFunction without reading comments... they seem the same to me.